src/dispatchers/pageDispatcher.ts

```diff
@@ -38,13 +38,21 @@ import { Download } from '../server/download';
 export class PageDispatcher extends Dispatcher<Page, channels.PageInitializer> implements channels.PageChannel {
   private _page: Page;
 
+  private static from(scope: DispatcherScope, page: Page | undefined): PageDispatcher | undefined {
+    if (!page)
+      return undefined;
+    const result = existingDispatcher<PageDispatcher>(page);
```

Comment:
There must be a dispatcher for opener already, right? Perhaps we don't need to create a new one here, just lookup?

Reply:
I suspect so, but I couldn't prove it strictly so I'd rather stay on the safe side and keep current logic.